DDF-1074 cache mgmt

wmcnalli opened the pull request:
@pklinef 
@kcwire 
@jlcsmith 
@emanns95
@jrnorth

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf-catalog/98)
<!-- Reviewable:end -->

jlcsmith commented:
<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java, line 43 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jrt0wOMoNTfO5MtSiFz)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java#L43)):</sup>
You can use java.util.concurrent.TimeUnit instead.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

jlcsmith commented:
<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java, line 43 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jrt0wOMoNTfO5MtSiFz)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java#L43)):</sup>
 TimeUnit.SECONDS.toMillis(1)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

jlcsmith commented:
<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 44 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jrt2KEXfuUSfdT1VL1j)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L44)):</sup>
Maybe use StringUtils.isEmpty() to account for null or an empty string.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

jlcsmith commented:
<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 50 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jrt3C7DKG0f1G-_M39s)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L50)):</sup>
This may be out of scope for this ticket, but wouldn't it be pretty straight forward to accept a CQL filter?  I am guessing the CQL support in the other catalog commands could be reused, but I might be wrong.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

kcwire commented:
Review status: 0 of 14 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java, line 52 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jry8XPQ0bE4_QVHDes_)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java#L52)):</sup>
I take it you can't make this abstract since it's coming from OsgiCommandSupport?

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 37 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jry9NXB7P1FgzwAux3-)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L37)):</sup>
If I'm not mistaken this is querying solr directly, so an admin would have to know the solr query syntax to build up a proper query.  It would be helpful if the description had a few basic examples.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

stustison commented:
@ wmcnalli Can you collapse all of these commits down to a single commit please?

jlcsmith commented on catalog-app/src/main/resources/features.xml:
```diff
@@ -52,6 +52,7 @@
         <bundle>mvn:ddf.catalog.core/catalog-core-urlresourcereader/${project.version}</bundle>
         <bundle>mvn:ddf.catalog.core/filter-proxy/${project.version}</bundle>
         <bundle>mvn:ddf.catalog.core/catalog-core-commands/${project.version}</bundle>
+	    <bundle>mvn:ddf.catalog.core/catalog-core-cachecommands/${project.version}</bundle>
```
Looks like the formatting is off here.

jlcsmith commented on core/catalog-core-standardframework/pom.xml:
```diff
@@ -199,7 +199,8 @@
                             ddf.catalog.cache.impl,
-                            ddf.catalog.cache.solr.impl,
+                            ddf.catalog.cache.solr,
+                            <!--ddf.catalog.cache.solr.impl, -->
```
remove if not neeed

stustison commented:
Review status: 0 of 14 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 63 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Js0jTlSu1J6KWppG3E0)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L63)):</sup>
You need to set the timezone on this to UTC

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

wmcnalli commented:
Will do.

---

Review status: 0 of 14 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java, line 52 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jry8XPQ0bE4_QVHDes_)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/CacheCommands.java#L52)):</sup>
Looks like I can abstract this class and method.  Fixed.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 37 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jry9NXB7P1FgzwAux3-)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L37)):</sup>
Fair enough.  I wrote a ticket to convert this over to CQL but I'll add a couple of examples in the meantime.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 50 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Jrt3C7DKG0f1G-_M39s)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L50)):</sup>
It's probably trivial but it was decided to save that for another ticket.  I only implemented this to make testing of the commands a little easier.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 63 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-catalog/98#-Js0jTlSu1J6KWppG3E0)** ([raw file](https://github.com/codice/ddf-catalog/blob/933c072d8c98a06d4c39a33d37fcfe5ad243d5f5/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L63)):</sup>

fixed

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

wmcnalli commented:
I hope I did that right.

---

Review status: 0 of 15 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/98)
<!-- Sent from Reviewable.io -->

kcwire commented:
+1 - if there no objections I'll merge this today

pklinef commented on core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java:
```diff
@@ -0,0 +1,76 @@
+    private static final int TITLE_MAX_LENGTH = 50;
+
+    @Argument(name = "SEARCH_PHRASE", index = 0, multiValued = false, required = false,
+            description = "Phrase to query the cache. \nExample:\n \"*:*\" will return all records in the cache. \n \"title:abc\" will return records with title = abc")
```
Probably should make this more clear that this is really a Solr query and not just a search phrase like in the catalog commands.  No one is really going to know the Solr index fields names either.  Not sure if we need a command to show those.

It might be more user friendly to leave this as a search phrase and just search the string given against the metadata field.

pklinef replied:
Another option would be to keep this solr query search for power admins but add another command that just queries the metadata and potentially another to search by title.

wmcnalli replied:
I wrote DDF-1262 to implement the CQL searching and I'll keep this lower-level functionality available with a flag.
